pilot/pkg/serviceregistry/kube/controller.go

```diff
@@ -715,6 +715,10 @@ func (c *Controller) AppendServiceHandler(f func(*model.Service, model.Event)) e
 		portsByNum := map[uint32]string{}
 
 		for _, port := range svc.Spec.Ports {
+			// Ignore udp svc port
+			if port.Protocol == v1.ProtocolUDP {
+				continue
```

Comment:
we also ignore udp in cluster.go and in listener.go. So how will this make a difference?

Reply:
I think the problem is that in some parts of the code we get UDP when on same name/port there is a TCP port as well.
This will just not add UDP - making the other code that ignores UDP safe ( since it won't accidentally ignore both TCP and UDP ). Or at least that's what I guess from the fact that this change appears to work, and without it it doesn't :-)

Reply:
( assuming this is the same with the Autotrader bug we're investigating) 

@howardjohn

Reply:
This did not fix the autotrader bug (the one linked in the description) from my testing

The reason this is different than just ignoring though:
The problem with kube-dns is it has two ports on port 53 - one for tcp and one for udp. So if we have a map of portnumber -> something, then only one of them can fit in this slot. So if the UDP one takes the slot, then we later filter it out, we are missing the TCP one.